### PR TITLE
feat(eigen-client-extra-features): Add option to download points

### PR DESCRIPTION
**Pull request by gianbelinche**
## What ❔
This PR adds an option to download points files instead of using a path

## Checklist

- [ ] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zkstack dev fmt` and `zkstack dev lint`.

### CODE REVIEW

**Comment by juan518munoz** [comment]
There are some `check` and `clippy` warnings, be sure to check them before merging

File: core/lib/config/src/configs/da_client/eigen.rs [hunk 1]
```diff
@@ -1,5 +1,17 @@
 use serde::Deserialize;
 use zksync_basic_types::secrets::PrivateKey;
+
+#[derive(Clone, Debug, PartialEq, Deserialize)]
+pub enum Points {
```
Comment (juanbono) [inline core/lib/config/src/configs/da_client/eigen.rs:5]: we should use a better name for this one

**Comment by gianbelinche** [comment]
> See comments

Done

File: core/lib/protobuf_config/src/da_client.rs [hunk 2]
```diff
@@ -73,9 +76,17 @@ impl ProtoRepr for proto::DataAvailabilityClient {
...
+                points_source: match conf.points_source.clone() {
+                    Some(proto::eigen_config::PointsSource::Path(path)) => {
+                        let path = required(&path.path).context("path")?;
+                        zksync_config::configs::da_client::eigen::PointsSource::Path(path.clone())
+                    }
+                    Some(proto::eigen_config::PointsSource::Link(link)) => {
+                        let link = required(&link.link).context("link")?;
+                        zksync_config::configs::da_client::eigen::PointsSource::Link(link.clone())
+                    }
+                    None => return Err(anyhow::anyhow!("Invalid Eigen DA configuration")),
```
Comment (juanbono) [inline core/lib/protobuf_config/src/da_client.rs:88]: why Anyhow instead of adding a proper error type?

Comment (gianbelinche) [reply core/lib/protobuf_config/src/da_client.rs:88]: That is how it is done for the other clients

**Review by juanbono — APPROVED** [review]

**Pull request merged**